code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -32,36 +40,19 @@
 		return null
 
 	var/atom/T = null
-	var/mob/living/TL = null
-	var/lowest_health = INFINITY // Max you can get
-	stop_automated_movement = 0
-	for(var/atom/A in targets)
+	for (var/atom/A in targets)
 		if(A == src)
 			continue
-
-		if (TL)
-			var/mob/living/L = A
-			if((L.faction == src.faction) && !attack_same)
-				continue
-			if(L in friends)
-				continue
-			if(!L.stat && (L.health < lowest_health))
-				lowest_health = L.health
-				T = L
-		else if (isliving(A))
-			TL = A
-
-		else if(istype(A, /obj/mecha)) // Our line of sight stuff was already done in ListTargets().
-			var/obj/mecha/M = A
-			if (M.occupant)
-				T = M
+		if (target_type_validator_map[A.type])

```

Comment:
Also I just realized, this doesn't cover subtypes. Which would be a logic error. Might be worth using this logic instead:
```dm
		if(A == src)
			continue

		var/datum/callback/cb = null
		for (var/type in target_type_validator_map)
			if (istype(A, type)
				cb= target_type_validator_map[type]

		if (!cb)
			continue
		else if (!istype(cb) || cb.Invoke(A))
			T = A
			break
```